libraries/AP_RangeFinder/AP_RangeFinder_NRA24_CAN.cpp

```diff
@@ -0,0 +1,86 @@
+#include "AP_RangeFinder_NRA24_CAN.h"
+#include <AP_BoardConfig/AP_BoardConfig.h>
+
+#if AP_RANGEFINDER_NRA24_CAN_ENABLED
+
+#include <AP_HAL/AP_HAL.h>
+
+RangeFinder_MultiCAN *AP_RangeFinder_NRA24_CAN::multican_NRA24;
+
+// constructor
+AP_RangeFinder_NRA24_CAN::AP_RangeFinder_NRA24_CAN(RangeFinder::RangeFinder_State &_state, AP_RangeFinder_Params &_params) :
+    AP_RangeFinder_Backend_CAN(_state, _params)
+{
+    if (multican_NRA24 == nullptr) {
+        multican_NRA24 = new RangeFinder_MultiCAN(AP_CAN::Protocol::NanoRadar_NRA24, "NRA24 MultiCAN");
+        if (multican_NRA24 == nullptr) {
+            AP_BoardConfig::allocation_error("Rangefinder_MultiCAN");
+        }
+    }
+
+    {
+        // add to linked list of drivers
+        WITH_SEMAPHORE(multican_NRA24->sem);
+        auto *prev = multican_NRA24->drivers;
+        next = prev;
+        multican_NRA24->drivers = this;
+    }
+}
+
+// update the state of the sensor
+void AP_RangeFinder_NRA24_CAN::update(void)
+{
+    WITH_SEMAPHORE(_sem);
+    if (get_reading(state.distance_m)) {
+        // update range_valid state based on distance measured
+        state.last_reading_ms = AP_HAL::millis();
+        update_status();
+    } else if (AP_HAL::millis() - state.last_reading_ms > read_timeout_ms()) {
+        if (AP_HAL::millis() - last_heartbeat_ms < read_timeout_ms()) {
+            // don't have distance data but sensor is connected. This is a known issue when sensor is in static condition (example Copter waiting to take off). Set status to out of range to avoid pre arm error
+            set_status(RangeFinder::Status::OutOfRangeLow);
```

Comment:
I'd prefer a flag to say that we allow arming with no data for this lidar, and use no data is we have not received any data for a period of time (1s?)

Reply:
Fixed. It didn't need a flag though as we have way to handle sensor specific pre arm check